packages/thrift-server-core/src/main/Int64.ts

```diff
@@ -87,55 +266,48 @@ export class Int64 {
         } else {
             // Most significant (up to 5) digits
             const high5 = +text.slice(negative ? 1 : 0, -15)
-            let low: number = +text.slice(-15) + high5 * 2764472320 // The literal is 10^15 % 2^32
-            let high: number = Math.floor(low / POW2_32) + high5 * 232830 // The literal is 10^15 / 2^&32
-            low = low % POW2_32
+            const remainder = +text.slice(-15) + high5 * 2764472320 // The literal is 10^15 % 2^32
+            const hi = Math.floor(remainder / POW2_32) + high5 * 232830 // The literal is 10^15 / 2^&32
+            const lo = remainder % POW2_32
 
             if (
-                high >= POW2_31 &&
-                !(negative && high === POW2_31 && low === 0) // Allow minimum Int64
+                hi >= POW2_31 &&
+                !(negative && hi === POW2_31 && lo === 0) // Allow minimum Int64
             ) {
                 throw new RangeError('The magnitude is too large for Int64.')
             }
 
             if (negative) {
-                // 2's complement
-                high = ~high
-
-                if (low === 0) {
-                    high = (high + 1) & 0xffffffff
-                } else {
-                    low = ~low + 1
-                }
-
-                high = 0x80000000 | high
+                const neg = hiLoNeg({ hi, lo })
+                return new Int64(neg.hi, neg.lo)
             }
 
-            return new Int64(high, low)
+            return new Int64(hi, lo)
         }
     }
 
+    /** @inheritDoc */
     public readonly buffer: Buffer
-    private offset: number
 
     constructor(buf: Buffer | Uint16Array | number, offset?: number)
     constructor(str: string)
     constructor(...args: Array<any>) {
         if (args[0] instanceof Buffer) {
-            this.buffer = args[0]
-            this.offset = args[1] || 0
+            const source: Buffer = args[0]
+            const offset: number = args[1] || 0
+            this.buffer = prepBuffer(source, offset)
```

Comment:
Only some methods handled the offset correctly. In addition, the protocols serialized the `Int64` using the buffer with no offset. Adjust for the offset in the constructor instead of trying to fix offset handling in all the methods and protocols. Making buffer private also seems like a breaking change to the public interface. Making offset public would allow for callers to handle the offset but makes all the callers more complicated (and does not fix existing callers).